pkg/api/v1alpha1/cluster_types.go

```diff
@@ -87,6 +93,18 @@ func (n *Cluster) Equal(o *Cluster) bool {
 	if n.Spec.KubernetesVersion != o.Spec.KubernetesVersion {
 		return false
 	}
+
+	switch n.Spec.DatacenterRef.Kind {
```

Comment:
Can we move this to its own method and add it as part of the return??

```
return n.Count == o.Count && n.MachineGroupRef.Equal(o.MachineGroupRef) &&
		TaintsSliceEqual(n.Taints, o.Taints) && MapEqual(n.Labels, o.Labels) && CPEndpointEqual()
```

Reply:
Do you mean add to `func (n *ControlPlaneConfiguration) Equal(o *ControlPlaneConfiguration) bool {`, and pass datacenterkind there?